File: Data Structures & Algorithms/sort-an-array/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    sortArray(nums) {
        let min = Math.min(...nums);
        let max = Math.max(...nums);

        let bucket  = Array.from({length :  max - min  + 1},() => 0);
        const res = [];
        for(let i = 0 ; i < nums.length;i++){
            bucket[nums[i] - min]++;
        }
    console.log(bucket)
        for(let i = 0 ; i < bucket.length;i++){
            let y = 0 ;
            while(y !== bucket[i]){
                res.push(i + min);
                y++;
            }
        }
        return res;


    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.go
func lengthOfLongestSubstring(s string) int {
	set := make(map[byte]bool)
	var l = 0
	var res = 0
	for r := 0 ; r < len(s); r++{
		for set[s[r]] {
            delete(set, s[l])
            l++
        }
		set[s[r]] = true

		res = max(res, r-l+1)
	}
	
	return res
}
